server/routes/user.js

```diff
@@ -72,28 +75,53 @@ router.post('/courses', verifyAuth, async function (req, res, next) {
     }
 });
 
-//Remove a user from a course, sends the updated user
-router.delete('/courses/:courseId', verifyAuth, async function (req, res) {
+//Remove a user from a course, removes them from all groups that belong to that course, returns user data
+router.delete('/courses/:courseId', verifyAuth, async function (req, res, next) {
     const courseId = req.params.courseId;
     const userId = req.body.userId;
-    try {
-        const userDoc = await User.findByIdAndUpdate(
-            userId,
-            { $pull: { courses: courseId } },
-            { useFindAndModify: false, new: true }
-        )
+
+        // start new session for transaction
+        const session = await User.startSession();
+        session.startTransaction();
+        try {
+            const opts = { session };
+            const userDoc = await User.findByIdAndUpdate(
+                userId,
+                { $pull: { courses: courseId } },
+                { useFindAndModify: false, new: true }
+            )
             .populate({ path: 'courses', model: 'Course' })
             .catch(err => {
                 if (err.kind == 'ObjectId') {
-                    throw new BadRequest('Invalid Course ID');
+                    throw new BadRequest('Invalid userID');
```

Comment:
I'm not sure if you made this change or I did, but this error message should be 'Invalid Course ID' because we're getting the userID from the token but the courseId from the req.params is not guaranteed to be an actual objectId.